src/main/scala/com/fulcrumgenomics/commons/io/Io.scala

```diff
@@ -46,15 +47,29 @@ trait IoUtil {
 
   /** How large a buffer should be used when buffering operations. */
   def bufferSize: Int = 32 * 1024
+  /** The level of compression to use when writing compressed output. */
+  def compressionLevel: Int = 5
 
-  /** Creates a new InputStream to read from the supplied path. */
+  /** Creates a new InputStream to read from the supplied path. Automatically handles gzipped files. */
   def toInputStream(path: Path) : InputStream = {
-    val stream =  if (Files.isSameFile(path, Io.StdIn)) System.in else Files.newInputStream(path)
-    new BufferedInputStream(stream, bufferSize)
+    PathUtil.extensionOf(path) match {
+      case Some(".gz") | Some(".bgz") | Some(".bgzip") =>
+        new GZIPInputStream(Files.newInputStream(path), bufferSize)
+      case _ => {
+        val stream =  if (Files.isSameFile(path, Io.StdIn)) System.in else Files.newInputStream(path)
+        new BufferedInputStream(stream, bufferSize)
+      }
+    }
   }
 
-  /** Creates a new BufferedReader to read from the supplied path. */
-  def toOutputStream(path: Path) : OutputStream = new BufferedOutputStream(Files.newOutputStream(path), bufferSize)
+  /** Creates a new OutputStream to read from the supplied path. Automatically handles gzipped files. */
+  def toOutputStream(path: Path) : OutputStream = {
+      PathUtil.extensionOf(path) match {
+        case Some(".gz") => new GZIPOutputStream(Files.newOutputStream(path), bufferSize) { this.`def`.setLevel(compressionLevel) }
+        case _           => new BufferedOutputStream(Files.newOutputStream(path), bufferSize)
```

Comment:
@sstadick @tfenne 
1. should we handle block-compressed data here like we do in [fgbio](https://github.com/fulcrumgenomics/fgbio/blob/master/src/main/scala/com/fulcrumgenomics/util/Io.scala#L56)
2.  If not, then we'll need to keep that functionality there when that PR comes about.  As if not, should we remove support for `.bgz` and `.bgzip` above?  Gzip can handle decompression of blocked gzip, but then we have asymmetric support.

Reply:
In concept it would be nice to have block-gzip support in commons, but _not_ I think if that involves adding `htsjdk` as a dependency for commons, which I suspect it would.  I'm ok with the support being asymmetric.  There are plenty of formats we implement read support for but not write.

Reply:
Either direction ends up being a little awkward. Are we agreeing that asymmetric support is okay for right now?

Reply:
Asymmetric is fine. I agree, that i don’t want htsjdk as a dependency in commons.